packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -0,0 +1,429 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import { IDatabaseCluster } from './cluster-ref';
+import { IDatabaseInstance } from './instance';
+import { CfnDBProxy, CfnDBProxyTargetGroup } from './rds.generated';
+
+/**
+ * The kinds of databases that the proxy can connect to.
+ * This value determines which database network protocol the proxy recognizes when it interprets network traffic to
+ * and from the database.
+ * The engine family applies to MySQL and PostgreSQL for both RDS and Aurora.
+ */
+export enum ProxyEngineFamily {
+  /**
+   * MYSQL
+   */
+  MYSQL = 'MYSQL',
+
+  /**
+   * POSTGRESQL
+   */
+  POSTGRESQL = 'POSTGRESQL',
+}
+
+/**
+ * SessionPinningFilter
+ *
+ * @see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/rds-proxy.html#rds-proxy-pinning
+ */
+export enum SessionPinningFilter {
```

Comment:
Switch to an enum-like pattern that we have in other places in the CDK. This will allow users to specify new filters when they are launched by RDS but not supported yet by the CDK.

Something like -

```ts
export class SessionPinningFilter {
  public static readonly EXCLUDE_VARIABLE_SETS = new SessionPinningFilter('EXCLUDE_VARIABLE_SETS');

  public static custom(filterName: string) {
    return new SessionPinningFilter(filterName);
  }

  private constructor(public readonly filterName) {
  }
}
```